src/clj_jgit/porcelain.clj

```diff
@@ -476,7 +479,9 @@
   (-> (Git/cloneRepository)
       (.setURI uri)
       ^TransportCommand (.setCredentialsProvider *cred-provider*)
-      (.setTransportConfigCallback *transport-callback*)))
+      (.setTransportConfigCallback *transport-callback*)
+      (.setCredentialsProvider(UsernamePasswordCredentialsProvider. nil *ssh-key-passphrase*))
```

Comment:
This essentially overrides any `*cred-provider*` set by the user. Which I guess is fine if we document it but maybe it would be better to alter the `with-identity` macro to set `*cred-provider*` accordingly if `*ssh-key-passphrase` is provided. Then it should work transparently for all commands.

Reply:
Thanks, I see your point - I added another commit so that clone-cmd is unchanged and the credentials provider is set in with-identity.

Reply:
Thanks, looks good! There was a little issue with `:trust-all? true` and the `user-pass-provider` proxy impl in general, which I fixed earlier. I'll try to cut a new release soonish.